includes/class-modal-checkout.php

```diff
@@ -562,28 +585,43 @@ public static function order_review_fragments( $fragments ) {
 	}
 
 	/**
-	 * Adds an additional button to the Thank you page
+	 * Render markup at the end of the "thank you" view.
+	 *
+	 * @param WC_Order $order The order related to the transaction.
 	 *
 	 * @return void
 	 */
-	public static function render_checkout_after_success_markup() {
+	public static function render_checkout_after_success_markup( $order ) {
+		if ( self::is_newsletter_signup_available() ) {
+			self::render_newsletter_signup_form( $order );
+		} else {
+			self::render_after_success_button();
+		}
+	}
+
+	/**
+	 * Render markup at the end of the "thank you" view.
+	 *
+	 * @return void
+	 */
+	private static function render_after_success_button() {
 		// phpcs:disable WordPress.Security.NonceVerification.Recommended
-		if (
-			empty( $_REQUEST['modal_checkout'] ) ||
-			empty( $_REQUEST['after_success_behavior'] ) ||
-			empty( $_REQUEST['after_success_url'] )
-		) {
+		if ( empty( $_REQUEST['modal_checkout'] ) ) {
 			return;
 		}
 
-		$button_label = ! empty( $_REQUEST['after_success_button_label'] ) ? sanitize_text_field( wp_unslash( $_REQUEST['after_success_button_label'] ) ) : __( 'Continue browsing', 'newspack-blocks' );
+		$button_label = ! empty( $_REQUEST['after_success_button_label'] ) ? sanitize_text_field( urldecode( wp_unslash( $_REQUEST['after_success_button_label'] ) ) ) : __( 'Continue browsing', 'newspack-blocks' ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$url          = ! empty( $_REQUEST['after_success_url'] ) ? sanitize_text_field( urldecode( wp_unslash( $_REQUEST['after_success_url'] ) ) ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
```

Comment:
Nit: should this be `esc_url`?

```suggestion
		$url          = ! empty( $_REQUEST['after_success_url'] ) ? esc_url( urldecode( wp_unslash( $_REQUEST['after_success_url'] ) ) ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
```

Reply:
Actually escaping is done in the output code a few lines below, so that's redundant. Changed in a19bd5a265c41e1aa17e801e0bace8e228281a86